File: 0875-koko-eating-bananas/0875-koko-eating-bananas.js
/**
 * @param {number[]} piles
 * @param {number} h
 * @return {number}
 */

var calculateTotalHours = function(piles, speed){
    let totalH = 0;
    for (let bananas of piles) {
        totalH += Math.floor((bananas + speed - 1) / speed); // FIX
    }
    return totalH;
}

var minEatingSpeed = function(piles, h) {
    let maxPile = Math.max(...piles);
    
    let low = 1, high = maxPile;
    let ans = maxPile;

    while (low <= high) {
        let mid = Math.floor((low + high) / 2);
        let totalH = calculateTotalHours(piles, mid); // FIX

        if (totalH <= h) {
            ans = mid;
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }
    return ans;
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution(object):
    # Function to calculate total hours at given speed
    def calculateTotalHours(self, piles, speed):
        totalH=0
        for bananas in piles:
            totalH += (bananas + speed - 1) // speed
        return totalH
    def minEatingSpeed(self, piles, h):
        """
        :type piles: List[int]
        :type h: int
        :rtype: int
        """
        maxPile= max(piles)
        low, high = 1, maxPile
        ans = maxPile
        #Binary Search on answer space
        while(low<=high):
            mid = (low + high) // 2
            totalH = self.calculateTotalHours(piles, mid)
             # If possible, try smaller speed
            if totalH <= h:
                ans = mid
                high = mid - 1
            # Otherwise, try larger speed
            else:
                low = mid + 1

        return ans
        